x/auth/keeper/grpc_query.go

```diff
@@ -25,11 +25,21 @@ func (ak AccountKeeper) AccountAddressByID(c context.Context, req *types.QueryAc
 	}
 
 	if req.Id < 0 {
-		return nil, status.Error(codes.InvalidArgument, "invalid account number")
+		return nil, status.Errorf(codes.InvalidArgument, "invalid account number %d", req.Id)
+	}
+
+	var accId uint64
+	switch {
+	case req.AccountId > 0 && req.Id > 0 && req.AccountId != uint64(req.Id):
```

Comment:
TBH I'm not sure how much value this case adds. IMO we should instead enforce one or the other value is passed. If both are passed, i.e. both are non-zero, then simply error.

Reply:
> IMO we should instead enforce one or the other value is passed

Unforunately we can't do this since `0` is also a valid case here for `account-number`. and the default values for both `int64` & `uint64` are `0`s

Reply:
We start account numbers at 1?

Reply:
I'm not sure, but I'm getting the response for `0` too by using the same query.

Reply:
Yeah I see that they start with `0`. It' really unfortunate we have to have this sort of case. I would almost prefer to completely omit/ignore the legacy field all together.

Reply:
I also agree with ignoring the legacy field. Maybe we can throw an error if it's not zero? So that users have immediate feedback they need to change something on their side.

Reply:
we are keeping it as deprecated means, may be it should not throw error unitl the field gets removed right? (not sure just a thought)

Reply:
The question really is:

Are we backporting this to < 0.47.x? If not, then just ignore the legacy field and remove this case all together. If we are backporting to 0.46.x and 0.45.x, then we need to handle the use of the legacy field and I suppose this is fine.

Reply:
I vote no backport.

Reply:
The problem is a zero/default value `0` is technically valid (the very first account). So there is no notion of "set" vs "unset" unfortunately. Thats why I suggested ignoring it.